proposals/0389-dynamic-min-deposit.md

```diff
@@ -0,0 +1,199 @@
+    min_deposit_next = min_deposit_current * 0.95   # -5%
+else:
+    min_deposit_next = min_deposit_current          # hold
+```
+
+Notes:
+
+- No proportional term; the controller is integral-only with a deadband. The
+  goal is to keep price pinned at the floor (0.1x) under normal conditions and
+  react only to pronounced spikes.
+- `H` (deadband threshold) is a protocol constant (see Protocol constants).
+- `I` is the integral accumulator over error; large positive values indicate
+  sustained overshoot; large negative values indicate sustained slack.
+
+Rationale: the main goal of the controller is to supervise state growth and
+engage effectively and predictably when the safety threshold is violated.
+
+- Asymmetric updates: bias towards upwards adjustments after engagement is
+  triggered allows a faster response to excessive state growth and a more
+  gradual return to baseline when conditions normalize.
+- Asymmetric integral bounds: the absolute value of `I_min` being larger than
+  `I_max` allows for the accumulation of a buffer to prevent smaller spikes
+  from causing a `min_deposit` adjustment, thereby increasing stability of the
+  0.1x floor price. A return to baseline can happen faster after a high
+  allocation event has passed, with the lower `I_max` value.
+- Deadband: avoids oscillations around the safety threshold.
+
+**Clamps:**
+
+Apply clamps after the update to ensure `min_deposit` stays within bounds:
+
+```
+min_deposit_next = clamp(
+  0.1 * min_balance_legacy,
+  1.0 * min_balance_legacy,
+  min_deposit_next
+)
+```
+
+where `min_balance_legacy` is the fixed minimum-balance constant used prior to
+this feature activation.
+
+### Sysvar exposure
+
+Expose `min_deposit` via the existing Rent sysvar to avoid introducing a new
+sysvar and to preserve compatibility with existing on-chain programs.
+
+### Protocol constants and feature gates
+
+- New constants:
+  - `G_target = 1 GiB/epoch`
+  - `H` deadband threshold for integral accumulator (tunable)
+  - `I_min = -4_000_000_000`, `I_max = +2_000_000_000`
+  - `down_step = -5%`, `up_step = +10%`
+  - `min_deposit_lower_bound = 0.1 * min_balance_legacy`
+  - `min_deposit_upper_bound = 1.0 * min_balance_legacy`
+
+## Alternatives Considered
+
+- Keep a fixed global constant: simple but provides no feedback control and
+  limited ability to respond to demand shocks.
+- Proportional-only controller: reacts quickly but adjustments are more erratic;
+  integral term addresses bias and long-term trends.
+- Hard step adjustments per epoch: simpler but introduces large
+  discontinuities and gaming incentives.
+- Dynamic rent w/ eviction: more complete but controversial and
+  complicated. This proposal will allow a reduction in state allocation cost
+  sooner.
+
+## Impact
+
+- Dapp developers: Creation and reallocation costs vary over time; programs can
+  query the Rent sysvar for current rates. Significant minimum-deposit
+  reduction is expected.
```

Comment:
I've heard that variable costs can be a pain. I don't have useful numbers for how much is too much though. This impacts regular users too, not just dapp developers.

This in turn impacts RPC calls and the CLI, which return the rent amount. 

Not saying these are deal breakers, just noting them.